### DISCUSSION TITLE
Simplified stack version needed

### DISCUSSION THREAD

**Discussion by sokoow** [tinkerbell/tink#457] [Ideas]
I'd love to run tinkerbell over the internet to help my small project (https://devpost.com/software/help-education-sector-to-produce-more-laptops-for-students), with significantly reduced stack components. 

I've a vision to run a 3-4 fixed, catch-all MACs workflows. Here's how I see it:

- get rid of most of logic for TFTP/DHCP/BOOTP - so no boots, cacher, and hegel needed in my case
- keep osie's tree - especially kernels and vmlinuzez, great concept
- keep the way you describe workflow steps in a singe template file - but somehow detach that logic from components I've cut down, another great concept

From my analysis, I conclude that a potential killer feature here to make this all happen would be... for worker helper/container to be able to read workflow definition from a https source. Would you think it through please and point me how this potentially could be implemented (I can try myself)?

I'm open to discuss this idea further if you wanted to.

**Comment by mmlb** [comment]
Hey @sokoow, welcome! Great idea you've got there, I'd love to see tink used there.

Why wouldn't gRPC work for you? We're working on "catch-all" MACs worflow support ourselves too, stay tuned for updates.

**Comment by sokoow** [comment]
mostly for simplicity, I'm orienting myself on a pull paradigm from some publicly available https source - who knows, maybe grpc would work here too. greetings for Zac by the way also :)

**Comment by thebsdbox** [comment]
At the moment the micro-services are quite tightly coupled, there would be a lot of effort required to perhaps split these out. In order to simplify these sorts of ideas we now have a proposals process that can raise these ideas for discussion available here -> https://github.com/tinkerbell/proposals
